packages/truffle-db/src/workspace/test/index.ts

```diff
@@ -249,274 +276,221 @@ mutation AddCompilation($compilerName: String!, $compilerVersion: String!, $sour
+  contractsAdd(input: {
+    contracts: [{
+      name: $contractName
+      abi: {
+        json: $abi
+      }
+      compilation: {
+        id: $compilationId
+      }
+      sourceContract: {
+        index: 0
+      }
+      constructor: {
         createBytecode: {
           id: $bytecodeId
         }
-        compilation: {
-          id: $compilationId
+      }
+    }]
+  }) {
+    contracts {
+      id
+      name
+      sourceContract {
+        name
+        source {
+          contents
         }
-        contract: {
-          id: $contractId
+        ast {
+          json
         }
-      }]
-    }) {
-      contractConstructors {
-        id
+      }
+      constructor {
         createBytecode {
           bytes
-          sourceMap
-        }
-        compilation {
-          compiler {
-            name
-          }
-          contracts {
-            name
-          }
-          sources {
-            sourcePath
-          }
-        }
-        contract {
-          name
-          source {
-            sourcePath
-          }
         }
       }
     }
+  }
 }`
 
-it("adds contract constructors", async () => {
-  const workspace = new Workspace();
+describe("Contract", () => {
+  const client = new WorkspaceClient();
 
   const variables = {
-    bytecodeId: soliditySha3(Migrations.bytecode), 
-    compilationId: "0xfd63faebf4e7515b2f48fdf22066869e09df9b2e91f4e0cf88d7d14ec496b515",
-    contractId: soliditySha3(Migrations.contractName, soliditySha3(Migrations.source, Migrations.sourcePath))
+    contractName: Migrations.contractName,
+    compilationId: compilationId,
+    id: contractId,
+    compilerName: Migrations.compiler.name,
+    compilerVersion: Migrations.compiler.version,
+    sourceId: sourceId,
+    bytecodeId: bytecodeId, 
+    abi: JSON.stringify(Migrations.abi)
   }
 
-  const constructorId = soliditySha3(variables.bytecodeId);
-
-  // add contract constructors
-  {
-    const result = await graphql(
-      schema, AddContractConstructors, null, { workspace }, variables
-    );
-
-    const { data } = result;
-    expect(data).toHaveProperty("contractConstructorsAdd");
-
-    const { contractConstructorsAdd } = data;
-    expect(contractConstructorsAdd).toHaveProperty("contractConstructors");
-
-    const { contractConstructors } = contractConstructorsAdd;
-
-    expect(contractConstructors).toHaveLength(1);
-
-    const { id } = contractConstructors[0];
-    expect(id).toEqual(constructorId);
+  beforeAll(async () => {
+    await client.execute(AddCompilation, variables);
+  });
+
+  it("adds contracts", async () => {
+    const client = new WorkspaceClient();
+
+    const variables = {
+      contractName: Migrations.contractName,
+      compilationId: compilationId,
+      id: contractId,
+      compilerName: Migrations.compiler.name,
+      compilerVersion: Migrations.compiler.version,
+      sourceId: sourceId,
+      bytecodeId: bytecodeId, 
+      abi: JSON.stringify(Migrations.abi)
+    }
```

Comment:
(This overwrites `variables` from the describe block... I think we should get rid of one of these declarations, either this one or the one above.)